packages/elements/list/src/getListInsertFragment.ts

```diff
@@ -1,51 +1,83 @@
 import { findNode } from '@udecode/plate-common';
 import {
   getPlatePluginOptions,
+  PlatePluginOptions,
   SPEditor,
   TDescendant,
 } from '@udecode/plate-core';
-import { Path, Transforms } from 'slate';
+import { Node, NodeEntry, Path, Transforms } from 'slate';
 import { ELEMENT_LI, ELEMENT_OL, ELEMENT_UL } from './defaults';
 
 export const getListInsertFragment = (editor: SPEditor) => {
   const { insertFragment } = editor;
 
-  return (fragment: TDescendant[]) => {
-    const li = getPlatePluginOptions(editor, ELEMENT_LI);
-    const ul = getPlatePluginOptions(editor, ELEMENT_UL);
-    const ol = getPlatePluginOptions(editor, ELEMENT_OL);
+  const li = getPlatePluginOptions(editor, ELEMENT_LI);
+  const ul = getPlatePluginOptions(editor, ELEMENT_UL);
+  const ol = getPlatePluginOptions(editor, ELEMENT_OL);
 
-    const liEntry = findNode(editor, { match: { type: li.type } });
+  const isListRoot = (node: TDescendant): boolean =>
+    [ul.type, ol.type].includes(node.type);
 
-    let filtered: TDescendant[] = [];
+  const getFirstAncestorOfType = (
+    root: TDescendant,
+    entry: NodeEntry,
+    { type }: PlatePluginOptions
+  ): NodeEntry<TDescendant> => {
+    let ancestor: Path = Path.parent(entry[1]);
+    while ((Node.get(root, ancestor) as TDescendant).type !== type) {
```

Comment:
Looks like a dangerous line for the undefined case, but looks to be ok for this scope. Needs a reminder if we plan to refactor it.

Reply:
There should be no undefined case - Path.parent should throw when there is no parent, so i believe this should be safe :thinking: Although, i didn't promote the function as i feel there is a better way about it.